processor/resourcedetectionprocessor/internal/resourcedetection_test.go

```diff
@@ -136,7 +136,7 @@ func TestDetectResource_Error(t *testing.T) {
 
 	p := NewResourceProvider(zap.NewNop(), time.Second, md1, md2)
 	_, _, err := p.Get(context.Background())
-	require.EqualError(t, err, "err1")
+	require.NoError(t, err)
```

Comment:
Can we add an error in TestMergeResource too to verify it produces the behavior we want?

Reply:
Hi @anuraaga in case of an error code flow won't execute the `MergeResource` function. So could you please elaborate more on what you are suggesting?

Reply:
Ah there's a test case just below TestMergeResource. It would be good to confirm we can add a failing detector but still pass that test.

Reply:
In the `TestMergeResource` test case we are taking two resources as input and using the `MergeResource` method to merge resources, But I don't see any use of detector in that testcase. So, how can we verify the failing detector in that test case and also we have covered the failing detector case in the `TestDetectResource_Error` test case by passing the `error` from the detector's `Detect` method.

Reply:
@rrupapara I think the request is for something like:

```
diff --git a/processor/resourcedetectionprocessor/internal/resourcedetection_test.go b/processor/resourcedetectionprocessor/internal/resourcedetection_test.go
index 8e5f62051..a14a37cfb 100644
--- a/processor/resourcedetectionprocessor/internal/resourcedetection_test.go
+++ b/processor/resourcedetectionprocessor/internal/resourcedetection_test.go
@@ -198,18 +198,23 @@ func TestDetectResource_Parallel(t *testing.T) {
 	md2 := NewMockParallelDetector()
 	md2.On("Detect").Return(NewResource(map[string]interface{}{"a": "11", "c": "3"}), nil)
 
+	md3 := NewMockParallelDetector()
+	md3.On("Detect").Return(pdata.NewResource(), errors.New("an error"))
+
 	expectedResource := NewResource(map[string]interface{}{"a": "1", "b": "2", "c": "3"})
 	expectedResource.Attributes().Sort()
 
-	p := NewResourceProvider(zap.NewNop(), time.Second, md1, md2)
+	p := NewResourceProvider(zap.NewNop(), time.Second, md1, md2, md3)
 
+	var detected pdata.Resource
 	// call p.Get multiple times
 	wg := &sync.WaitGroup{}
 	wg.Add(iterations)
 	for i := 0; i < iterations; i++ {
 		go func() {
 			defer wg.Done()
-			_, _, err := p.Get(context.Background())
+			var err error
+			detected, _, err = p.Get(context.Background())
 			require.NoError(t, err)
 		}()
 	}
@@ -220,11 +225,16 @@ func TestDetectResource_Parallel(t *testing.T) {
 	// detector.Detect should only be called once, so we only need to notify each channel once
 	md1.ch <- struct{}{}
 	md2.ch <- struct{}{}
+	md3.ch <- struct{}{}
 
 	// then wait until all goroutines are finished, and ensure p.Detect was only called once
 	wg.Wait()
 	md1.AssertNumberOfCalls(t, "Detect", 1)
 	md2.AssertNumberOfCalls(t, "Detect", 1)
+	md2.AssertNumberOfCalls(t, "Detect", 3)
+
+	detected.Attributes().Sort()
+	assert.Equal(t, expectedResource, detected)
 }
 
 func TestAttributesToMap(t *testing.T) {
```

Reply:
Sorry for the late reply @rrupapara and @rmfitzpatrick for creating the snippet I had in mind indeed